tests/unit/ui/components/TestConfirmButton.js

```diff
@@ -0,0 +1,140 @@
+  });
+
+  it('passes the children prop to the button', () => {
+    const children = 'this is the text of the button shown by default';
+    const root = render({ children });
+
+    expect(root.find(Button).children()).toHaveText(children);
+  });
+
+  it('shows a confirmation panel when button is clicked', () => {
+    const message = 'this action is risky, are you sure?';
+    const root = render({ message });
+
+    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
+
+    root.find(Button).simulate('click', createFakeEvent());
+
+    expect(root).toHaveClassName('ConfirmButton--show-confirmation');
+
+    expect(root.find('.ConfirmButton-message')).toHaveLength(1);
+    expect(root.find('.ConfirmButton-message')).toHaveText(message);
+
+    expect(root.find(Button)).toHaveLength(2);
+
+    const confirmButton = root.find(Button).at(0);
+    expect(confirmButton).toHaveClassName('ConfirmButton-confirm-button');
+    expect(confirmButton).toHaveProp('buttonType', 'alert');
+    expect(confirmButton.children()).toHaveText('Confirm');
+
+    const cancelButton = root.find(Button).at(1);
+    expect(cancelButton).toHaveClassName('ConfirmButton-cancel-button');
+    expect(cancelButton).toHaveProp('buttonType', 'cancel');
+    expect(cancelButton.children()).toHaveText('Cancel');
+  });
+
+  it('passes props to the confirm button in the confirmation panel', () => {
+    const confirmButtonText = 'neutral confirm button';
+    const confirmButtonType = 'neutral';
+    const root = render({ confirmButtonText, confirmButtonType });
+
+    // Show the confirmation panel.
+    root.find(Button).simulate('click', createFakeEvent());
+
+    expect(root.find('.ConfirmButton-confirm-button'))
+      .toHaveProp('buttonType', confirmButtonType);
+    expect(root.find('.ConfirmButton-confirm-button').children())
+      .toHaveText(confirmButtonText);
+  });
+
+  it('passes props to the cancel button in the confirmation panel', () => {
+    const cancelButtonText = 'neutral cancel button';
+    const cancelButtonType = 'neutral';
+    const root = render({ cancelButtonText, cancelButtonType });
+
+    // Show the confirmation panel.
+    root.find(Button).simulate('click', createFakeEvent());
+
+    expect(root.find('.ConfirmButton-cancel-button'))
+      .toHaveProp('buttonType', cancelButtonType);
+    expect(root.find('.ConfirmButton-cancel-button').children())
+      .toHaveText(cancelButtonText);
+  });
+
+  it('closes the confirmation panel on cancel ', () => {
+    const root = render();
+
+    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
+
+    // Show the confirmation panel.
+    root.find(Button).simulate('click', createFakeEvent());
+
+    expect(root).toHaveClassName('ConfirmButton--show-confirmation');
+
+    root.find('.ConfirmButton-cancel-button').simulate(
+      'click',
+      createFakeEvent()
+    );
+
+    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
```

Comment:
You have a test to check that `onConfirm` is called when the user clicks the "confirm" button, so maybe it would be worth adding an assertion into this test to assert that `onConfirm` is not called when a user clicks the "cancel" button?